### ISSUE TITLE
Param Synchronous seems to not work

### ISSUE THREAD

**Issue by Jean-MarieB** [meilisearch/meilisearch-rails#202]
Hi everyone,

**Description**
This issue is about synchronous feature, that can be set in `meilisearch` options in Model or in ms_index! parameters :

In model : 

```
# my_model.rb
meilisearch synchronous: false|true do
  # do something
end
```

With class_method:

```
MyModel.reindex!(MeiliSearch::Rails::IndexSettings::DEFAULT_BATCH_SIZE, true) # true is for param synchronous if I read meilisearch gem code correctly)
```

I want to use meilisearch synchronously for my test.

**Expected behavior**
I'm trying to test my search methods for my application. For example, if I'm searching for users named 'John' I test if 'John' is written, every users (in fixtures) with first name 'John' is found.

The searching tests are expected to work every time. I set meilisearch to synchronous to be sure that indexes are presents before search methods is call.
 

**Current behavior**
When I run tests `rails test`, searching tests failed randomly. Sometimes everything is ok, sometimes search on one model fail, sometimes is another etc etc.
I try to set synchronous to true in model, in `reindex!` class method, both, but it's still failing randomly.

Furthermore, I try to had some `sleep()` to force tests to wait some more seconds to help meilisearch to finish indexation first.

**Conclusion**
I don't know if I miss something or if I do a bad configuration of meilisearch. Thank you for your help 😄 

**Environment (please complete the following information):**
- Rails version: 7.0.3
- Meilisearch server (use with docker image) : v0.29.0
- meilisearch-rails version : 0.7.3
- Mac OS Monterey

**Comment by bdesouza** [comment]
Any idea where this landed? I'm having the same issue. Tests randomly fail due to slow response from meilisearch. If I step through them with a debugger they pass. Running each test individually also passes. FWIW, I'm using https://github.com/grosser/parallel_tests

**Comment by brunoocasali** [comment]
Hi @bdesouza and @Jean-MarieB I would not recommend you to e2e test the meilisearch requests, but mock them instead. It should solve most of your problems.

**Comment by brunoocasali** [comment]
Hey folks, I'm closing this issue. 

Although what you mentioned could be improved in the core, I don't see this as a concern that the team should spend time fixing. You can create a custom script that runs before every test to ensure no tasks are waiting to be processed before starting your suite. You can access the functions by `MyModel.index.tasks['results']` and iterate over them to find unprocessed ones. 
But again, I don't recommend testing the meilisearch behavior in your test like this. 
We do that in all the SDKs we maintain, and I can tell that is a pain 😅. 
For most use cases, ensuring that you're calling the right methods with the right parameters at the right time should be enough. 

If you have a particular test that you think should be nice to do that we're not doing here or in the meilisearch-ruby, please send a new issue so we can put it in the source for everybody to leverage that :)

Again, sorry if that was not what you expected to hear 😇 

Thanks a lot for using Meilisearch :)
